run-test.sh

```diff
@@ -55,6 +55,18 @@ usage()
     exit 1
 }
 
+# Handle Ctrl-C.
+function handle_ctrl_c {
+  local errorSource='handle_ctrl_c'
+
+  echo ""
+  echo "Trapped Ctrl-c event. It will stop running."
```

Comment:
I think the message could be a bit more friendly... something like "Cancelling test execution" as user's may not need to know about the mechanism by which the cancel happened.